File: scripts/check-postgres.js
#!/usr/bin/env node
/**
 * Check that Postgres is reachable (for start-sandarb.sh).
 * Loads .env from project root; defaults DATABASE_URL to local docker-compose URL.
 * Exits 0 if OK, 1 and message if not running / unreachable.
 */

const path = require('path');
const fs = require('fs');

const root = path.resolve(__dirname, '..');
const envPath = path.join(root, '.env');
if (fs.existsSync(envPath)) {
  const content = fs.readFileSync(envPath, 'utf8');
  for (const line of content.split('\n')) {
    const trimmed = line.replace(/^#.*/, '').trim();
    const m = trimmed.match(/^([A-Za-z_][A-Za-z0-9_]*)=(.*)$/);
    if (m) {
      const key = m[1];
      let val = m[2].replace(/^["']|["']$/g, '').trim();
      if (!process.env[key]) process.env[key] = val;
    }
  }
}

const DATABASE_URL = process.env.DATABASE_URL || 'postgresql://postgres:sandarb@localhost:5432/sandarb-dev';

const { Client } = require('pg');
const client = new Client({ connectionString: DATABASE_URL });

client
  .connect()
  .then(() => client.end())
  .then(() => {
    console.log('Postgres is running.');
    process.exit(0);
  })
  .catch((err) => {
    console.error('Postgres is not running or unreachable:', err.message);
    console.error('');
    console.error('Start Postgres (e.g. with Docker):');
    console.error('  docker compose up -d postgres');
    console.error('');
    console.error('Or set DATABASE_URL in .env to your Postgres connection string.');
    process.exit(1);
  });

File: scripts/check_postgres.py
#!/usr/bin/env python3
"""
Check that Postgres is reachable (for start-sandarb.sh).
Loads .env from project root; defaults DATABASE_URL to local docker-compose URL.
Exits 0 if OK, 1 if not running / unreachable.
"""
import sys
from pathlib import Path

# Scripts dir on path so we can import _env (same directory)
sys.path.insert(0, str(Path(__file__).resolve().parent))
from _env import load_dotenv, get_database_url

load_dotenv()


def main() -> None:
    try:
        import psycopg2
    except ImportError:
        print("psycopg2 not installed. pip install psycopg2-binary", file=sys.stderr)
        sys.exit(1)
    url = get_database_url()
    try:
        conn = psycopg2.connect(url)
        conn.close()
    except Exception:
        sys.exit(1)
    sys.exit(0)


if __name__ == "__main__":
    main()
